splunk_connect_for_snmp/common/task_generator.py

```diff
@@ -24,32 +32,44 @@ def generate_task_definition(self):
 
 
 class WalkTaskGenerator(TaskGenerator):
-
-    WALK_CHAIN_OF_TASKS = {
-        "link": chain(
-            signature("splunk_connect_for_snmp.enrich.tasks.enrich")
-            .set(queue="poll")
-            .set(priority=4),
-            group(
-                signature(
-                    "splunk_connect_for_snmp.inventory.tasks.inventory_setup_poller"
-                )
+    def __init__(
+        self,
+        target,
+        schedule_period,
+        app,
+        chain_of_tasks_expiry_time,
+        host_group,
+        profile,
+    ):
+        super().__init__(
+            target, schedule_period, app, chain_of_tasks_expiry_time, host_group
+        )
+        self.WALK_CHAIN_OF_TASKS = {
+            "expires": self.chain_of_tasks_expiry_time,
+            "link": chain(
+                signature("splunk_connect_for_snmp.enrich.tasks.enrich")
                 .set(queue="poll")
-                .set(priority=3),
-                chain(
-                    signature("splunk_connect_for_snmp.splunk.tasks.prepare")
-                    .set(queue="send")
-                    .set(priority=1),
-                    signature("splunk_connect_for_snmp.splunk.tasks.send")
-                    .set(queue="send")
-                    .set(priority=0),
+                .set(priority=4),
+                group(
+                    signature(
+                        "splunk_connect_for_snmp.inventory.tasks.inventory_setup_poller",
+                        kwargs={
+                            "chain_of_tasks_expiry_time": self.chain_of_tasks_expiry_time
```

Comment:
why this parameter is also in kwargs?

Reply:
This is because polling tasks are set up inside walk tasks. We need to somehow pass this parameter to the `inventory_setup_poller ` function, so that worker running this walk task knows what should be the `expiry` value for polling tasks.

I also tested the solution when we don't pass this parameter this way, instead we take it from the environmental variable inside a worker which sets up new polling tasks. The idea was that after this env variable is changed, the worker would restart with the new env variable value and new polling tasks would be created with updated `expiry` time. The problem was, that after running `helm upgrade` with env variable changed, that old worker still worked for a short period of time, and some of the records from the inventory didn't have `expiry` value changed while others had. 
So I decided that we will be checking `CHAIN_OF_TASKS_EXPIRY_TIME` env variable only in `loader` where it is always going to be up to date and we will pass this value to walk tasks. New polling tasks are later created inside walk task by the workers.

Reply:
after the newest changes - assuming we're always deleting all the polls and walks, can we just use env variable to set it up?